### ISSUE TITLE
[docs] authManager configuration for console

### ISSUE THREAD

**Issue by koxu1996** [yiisoft/yii2#19652]
**TL;DR** Why docs are advising to NOT configure `authManager` for console in advanced-app?

---

There is a quite popular [question about `authManager` error during rbac migration](https://stackoverflow.com/q/30478903/2582462), that has following answer (May 2015):

> In case of yii2-app-advanced the authManager should be declared in console/config/main.php and not in the backend or frontend configuration files.

**That solution seems to be working for me; previously I was randomly encountering above-mentioned error.**

However, year after that (Mar 2016) Yii2 documention is updated with this [note](https://github.com/yiisoft/yii2/commit/7667aa1ce28de0e248c20a1709eebcdfc3dabe42):

> If you are using yii2-basic-app template, there is a `config/console.php` configuration file where the `authManager` needs to be declared additionally to `config/web.php`. In case of yii2-advanced-app the `authManager` should be declared only once in `common/config/main.php`.

That makes me and [other people](https://stackoverflow.com/questions/30478903/yii2-cannot-perform-rbac-migration-you-should-configure-authmanager#comment64539615_30478904) confused, so if you have some idea what is correct configuration here, please let me know.

**Comment by terabytesoftw** [comment]
Thanks for the report.

The current documentation is correct for the advanced application template. In `yii2-app-advanced`, the console entry script builds the application configuration by merging `common/config/main.php` before `console/config/main.php`, so a component declared in `common/config/main.php` is available to the console application as well as to the web applications.

For `yii2-basic-app`, web and console configurations are separate, so `authManager` has to be configured in both `config/web.php` and `config/console.php` if both applications need it.

The StackOverflow answer referenced in the issue is therefore incomplete/outdated for the advanced template. Configuring `authManager` only in the console config may make RBAC migrations work, but it does not provide the component to frontend/backend. The recommended place for the advanced template is still `common/config/main.php`.

Closing this as answered
